backends/tfhe-cuda-backend/cuda/src/crypto/torus.cuh

```diff
@@ -27,6 +28,14 @@ __device__ inline void typecast_double_to_torus<uint64_t>(double x,
   r = lll;
 }
 
+template <typename T>
+__device__ inline void typecast_double_round_to_torus(double x, T &r) {
+  constexpr double mx = get_two_pow_torus_bits<T>();
+  double frac = x - floor(x);
```

Comment:
this is a round not a floor in the CPU implementation

Reply:
@IceTDrinker  is there a specific reason why round can work better than floor here? Are there any benefits?

Reply:
also round for us does not work tests are failing

Reply:
I think here it may not have that big of an impact as it would essentially flip the sign I believe, as frac should end up between -0.5 and 0.5, in some cases I recall rounding towards 0 or rounding towards infinity as having a stabilizing effect on float computations (but then the ranges are really not the same)

Reply:
really ? interesting, which tests are failing ?

Reply:
pbs tests

Reply:
ok so it means there is something about correctness with the proper rounding mode, I'll check for rust as the rounding mode may be a bit peculiar for floats, please add a comment about the correctness of floor so we don't hesitate in the future, I'll add an issue for the CPU implem and check if floor would be preferable to round in the implem

Reply:
weird actually it should not affected it floor on minus values should be same as round on positive ones maybe  
`.5` is not acting as expected